core/src/main/scala/kafka/log/ProducerIdMapping.scala

```diff
@@ -261,7 +261,7 @@ class ProducerIdMapping(val config: LogConfig,
         case None =>
           lastSnapOffset = 0L
           lastMapOffset = 0L
-          snapDir.mkdir()
+          Files.createDirectories(snapDir.toPath)
```

Comment:
It seems to me that we only need to recreate the directory if it gets deleted by some code after the constructor of `PidMapping` because the path to the directory does not change. Is that the case?